### DISCUSSION TITLE
Ability to reload the config file upon changes to the config

### DISCUSSION THREAD

**Discussion by wacns** [thejoabo/virtualfisher-bot#132] [Ideas]
Hello, I would like to suggest a feature that is about reloading the config details upon any change to the running config file and resuming the process with the new config. I've tried to do it but had no enough knowledge about how to do it due to the complexity of how the program is coded.

**Comment by wacns** [comment]
I've achieved this goal without any modification to the main process **(I'm not sure if this concept is efficient but it does the job by terminating the current process and start a new process)**:
```python
import subprocess
import time
from watchdog.observers import Observer
from watchdog.events import FileSystemEventHandler


class RestartOnChangeHandler(FileSystemEventHandler):
    def __init__(self, command):
        self.command = command
        self.pid: subprocess.Popen[bytes]
    
    def start(self):
        self.pid = subprocess.Popen(self.command)

    def on_any_event(self, event):
        print(f"Change detected: {event.src_path}. Restarting script...")
        self.pid.terminate()
        self.pid = subprocess.Popen(self.command)

def main(script_path):
    path = './configs'
    command = ['python', script_path]

    event_handler = RestartOnChangeHandler(command=command)
    observer = Observer()
    observer.schedule(event_handler, path, recursive=True)
    observer.start()

    print(f"Monitoring {path} for changes. Running {script_path}...")
    event_handler.start()

    try:
        while True:
            time.sleep(1)
    except KeyboardInterrupt:
        observer.stop()

    observer.join()

if __name__ == "__main__":
    main('autofishbot.py')
```

**Reply by thejoabo** [reply]
Hello @wacns , interesting idea. I never thought about a hot-reload feature for config files. This is indeed a viable solution and would work well, but I need to check how it affects the multiple threads running different parts of the bot logic. Mostly, debugging is needed to confirm that it truly doesn't change the behavior at all.

I really apologize that I couldn't spend more time working on this project and actually engaging in its development. Anyways, I'm gathering all these ideas for a future refactor (I'm planning to migrate this project to Typescript + Node for many reasons), so any major changes using Python seem a bit problematic. Nevertheless, I really appreciate your idea and contribution.

**Reply by wacns** [reply]
Hi Joabo, I'm trying to make some modifications to the scheduler class for example adding /use Personal and few more commands but seems to have problem with timing can you reply to my email that I've sent to help me achieving that?

**Comment by wacns** [comment]
As well as if you can implement the ability to retrieve the cooldown during the fishing session as within some points you get cooldown message including the current cooldown so you can compare it with the current cooldown
![image](https://github.com/thejoabo/autofishbot/assets/43343208/e004bb27-4d6b-4bc4-9e5f-0ac7b0cb0cf1)
